addon/components/one-way-input.js

```diff
@@ -34,10 +34,23 @@ export default Component.extend({
     'value',
     'width'
   ],
+  KEY_EVENTS: {
+    '13': 'onenter',
+    '27': 'onescape'
+  },
   _sanitizedValue: undefined,
 
   input() { this._handleChangeEvent(); },
   change() { this._handleChangeEvent(); },
+  keyUp(event) { this._interpretKeyEvents(event); },
+
+  _interpretKeyEvents(event) {
+    const method = this.KEY_EVENTS[event.keyCode];
+
+    if (method) {
+      return this.attrs[method](this.sanitizeInput(this.readDOMAttr('value')));
```

Comment:
If this.attrs.onenter or this.attrs.onescape are not provided (perhaps because you don't need those functions) this will go BOOM.